Riot/Modules/Room/DataSources/RoomDataSource.m

```diff
@@ -963,4 +974,72 @@ - (void)bubbleReactionsViewModel:(BubbleReactionsViewModel *)viewModel didLongPr
     [self.delegate dataSource:self didRecognizeAction:kMXKRoomBubbleCellLongPressOnReactionView inCell:nil userInfo:@{ kMXKRoomBubbleCellEventIdKey: eventId }];
 }
 
+#pragma mark - Room creation intro cell
+
+- (BOOL)canShowRoomCreationIntroCell
+{
+    NSString* userId = self.mxSession.myUser.userId;
+
+    if (!userId || !self.isLive || self.isPeeking)
+    {
+        return NO;
+    }
+    
+    // Room creation cell is only shown for the creator
+    return [self.room.summary.creatorUserId isEqualToString:userId];
+}
+
+- (void)enableRoomCreationIntroCellDisplayIfNeeded
+{
+    self.showRoomCreationCell = [self canShowRoomCreationIntroCell];
+}
+
+// Insert the room creation intro cell at the begining
+- (void)insertRoomCreationIntroCellDataIfNeeded
+{
+    if (self.showRoomCreationCell)
+    {
+        NSUInteger existingRoomCreationCellDataIndex = [self roomBubbleDataIndexWithTag:RoomBubbleCellDataTagRoomCreationIntro];
+        
+        if (existingRoomCreationCellDataIndex != NSNotFound)
+        {
+            [bubbles removeObjectAtIndex:existingRoomCreationCellDataIndex];
```

Comment:
`bubbles` is built from another thread. We need a lock like `@synchronized(bubbles)`